model/model.py

```diff
@@ -0,0 +1,109 @@
+import os
+import time
+import json
+import argparse
+import duckdb
+import numpy as np
+from datasketch import MinHash, MinHashLSH
+
+class KnowledgeGraph:
+    def __init__(self, directory: str):
+        self.conn = duckdb.connect()
+        self.conn.execute(
+        '''CREATE TABLE nodes (
+            id             DECIMAL(18, 0) NOT NULL PRIMARY KEY,
+            dbname         VARCHAR(255),
+            path           VARCHAR(255),
+            sourcename     VARCHAR(255),
+            columnname     VARCHAR(255),
+            datatype       VARCHAR(255),
+            totalvalues    INT,
+            uniquevalues   INT,
+            nonemptyvalues INT,
+            entities       VARCHAR(255),
+            minhash        BLOB,
+            minvalue       DECIMAL(18, 4),
+            maxvalue       DECIMAL(18, 4),
+            avgvalue       DECIMAL(18, 4),
+            median         DECIMAL(18, 0),
+            iqr            DECIMAL(18, 0)
+        )''')
+        self.conn.execute('''CREATE TABLE edges
+        (
+            from_node DECIMAL(18, 0) REFERENCES nodes(id),
+            to_node   DECIMAL(18, 0) REFERENCES nodes(id),
+            weight    DECIMAL(10, 4)
+        ) ''')
+        self.minhash_perm = None
+        # Populate the nodes table with JSON files under the specified path
+        for filename in os.listdir(directory):
+            filepath = os.path.join(directory, filename)
+            if os.path.isfile(filepath):
+                with open(filepath) as f:
+                    column = json.load(f)
+                    self.add_node(column)
+        
+        self.make_edges()
```

Comment:
would be great to factor this out into further functions, e.g.:

make_similarity_edges
make_neighbor_edges
make_pkfk_edges
etc

that way it'll be easier to extend later, if we need to (we will)

 Of course, all those functions will end up calling an add_edge() function, in the style you have right now, simply indicating the info needed to create the edge.